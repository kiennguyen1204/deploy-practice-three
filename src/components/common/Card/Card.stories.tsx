import { StoryObj, Meta } from '@storybook/react';
import Card from './index';

export default {
  title: 'Components/Card',
  component: Card
} as Meta;

type Story = StoryObj<typeof Card>;

export const Default: Story = {
  args: {
    classes: '',
    title: 'Example Card',
    iShowAll: true,
    children: 'Card Content'
  }
};
