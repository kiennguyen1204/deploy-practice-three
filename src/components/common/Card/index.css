.card-box-container {
  position: relative;
  width: 100%;
  height: auto;
}

.card-box-container .header-title {
  padding-bottom: 24px;
  margin: 24px;
  border-bottom: 1px solid var(--color-neutral-lightest);
}

.card-box-container .header-title .card-title {
  font-size: var(--font-size-xxl);
  line-height: var(--line-height-xl);
  letter-spacing: -0.05px;
  font-weight: var(--font-weight-medium);
  color: var(--color-neutral-base);
}

.card-box-container .header-title .show-all {
  color: var(--color-primary-base);
  text-decoration: underline;
  font-size: var(--font-size-medium);
}

.card-box-container .card-box {
  position: relative;
  padding: 0 24px;
  margin-bottom: 64px;
}

.card-box-container .card-box .card-box-element {
  position: relative;
  border-radius: 24px;

  color: var(--color-neutral-white);
  background: var(--color-secondary-dark);
  padding: 24px;
}

.show-all-link {
  color: var(--color-primary-base);
  font-size: var(--font-size-normal);
  font-weight: var(--font-weight-regular);
  line-height: var(--line-height-medium);
  text-decoration-line: underline;
}

@media screen and (min-width: 768px) {
  .card-box-container .card-box .card-box-element {
    padding: 64px;
  }
}

@media screen and (min-width: 992px) {
  .card-box-container .card-box {
    padding: 64px 120px;
  }

  .card-box-container .header-title {
    margin: 0 120px 64px 120px;
  }
}
