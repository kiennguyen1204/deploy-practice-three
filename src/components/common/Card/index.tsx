import React, { ReactNode, memo } from 'react';

// styles
import './index.css';

export interface Props {
  classes: string;
  title?: string;
  children: ReactNode;
}

const Card: React.FC<Props> = ({ children, classes, title = '' }): JSX.Element => (
  <article className={`card-box-container ${classes}`}>
    {title.length > 0 && (
      <header className='flex-space-between-container header-title'>
        <h2 className='card-title'>{title}</h2>

        <a href='https://weedmaps.com/learn' className='show-all-link'>
          Show All
        </a>
      </header>
    )}

    <section className='card-box'>
      <div className='card-box-element'>{children}</div>
    </section>
  </article>
);

export default Card;
