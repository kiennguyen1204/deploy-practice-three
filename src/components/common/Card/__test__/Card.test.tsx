import { render, RenderResult } from '@testing-library/react';
import Card from 'components/common/Card';

describe('Card', () => {
  let title: string, children: JSX.Element;

  const setup = (title: string | undefined, children: JSX.Element): RenderResult => {
    return render(
      <Card classes='test-class' title={title}>
        {children}
      </Card>
    );
  };

  beforeEach(() => {
    title = 'Test Title';
    children = <div>Test Children</div>;
  });

  it('renders the title and children', () => {
    const { getByText } = setup(title, children);

    expect(getByText(title)).toBeInTheDocument();
  });

  it('does not render the title when title is undefined', () => {
    const { queryByText } = setup(undefined, children);
    expect(queryByText(title)).not.toBeInTheDocument();
  });

  it('renders the card component correctly', () => {
    const { container } = setup(title, children);

    expect(container.firstChild).toMatchSnapshot();
  });
});
