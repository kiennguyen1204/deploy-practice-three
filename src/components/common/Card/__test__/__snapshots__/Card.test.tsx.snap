// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders the card component correctly 1`] = `
<article
  class="card-box-container test-class"
>
  <header
    class="flex-space-between-container header-title"
  >
    <h2
      class="card-title"
    >
      Test Title
    </h2>
    <a
      class="show-all-link"
      href="https://weedmaps.com/learn"
    >
      Show All
    </a>
  </header>
  <section
    class="card-box"
  >
    <div
      class="card-box-element"
    >
      <div>
        Test Children
      </div>
    </div>
  </section>
</article>
`;
